src/mlpack/methods/ann/layer/binary_rbm_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file binary_rbm_impl.hpp
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_BINARY_RBM_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_BINARY_RBM_IMPL_HPP
+// In case it hasn't yet been included.
+#include "binary_rbm.hpp"
+
+#include <mlpack/core/math/random.hpp>
+#include <mlpack/methods/ann/activation_functions/logistic_function.hpp>
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+VisibleLayer<InputDataType, OutputDataType>::VisibleLayer(
+  const size_t inSize,
+	const size_t outSize):
+  inSize(inSize), 
+  outSize(outSize)
+{
+  weights.set_size(outSize * inSize + inSize, 1);
```

Comment:
I did not understand your comment your saying that when we declare a new layer we do not allocate memory to it. Other layer allocate memory when they are created. Also we cannot this in the vanilla_rbm as we do not have acess to the insize and outsize there.